File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {

        StringBuffer sb = new StringBuffer(word1.length() + word2.length());
        int n = Math.min(word1.length(),word2.length());

        int i = 0;
        while(i < n)
        {
            sb.append(word1.charAt(i));
            sb.append(word2.charAt(i));
            i++;
        }

        if(n == word2.length())
        {
            sb.append(word1.substring(i));
        }

        else
        {
            sb.append(word2.substring(i));
        }

        return sb.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n;
        if(word1.size() > word2.size())
        {
            n = word2.size();
        }

        else
        {
            n = word1.size();
        }

        string res = "";
        int i = 0;
        while(i < n)
        {
            res += word1[i];
            res += word2[i];
            i++;
        }

        if(n == word2.size())
        {
            res += word1.substr(i);
        }

        else
        {
            res += word2.substr(i);
        }

        return res;
    }
};